compiler/qsc/src/error.rs

```diff
@@ -67,3 +68,18 @@ impl<E: Diagnostic> Diagnostic for WithStack<E> {
         self.error.diagnostic_source()
     }
 }
+
+pub fn eval(
```

Comment:
nitpick: the name `eval` makes it feel like this should be evaluating something, but really it's a conversion helper. Perhaps something like `from_eval`?

Reply:
I really wanted this to be `eval_error`, because, you know, it's creating an eval error, but Clippy shamed me for using the name of the module in the function. 

I really don't get along with Clippy.